### PR TITLE
Test file parser

### PR DESCRIPTION
**Pull request by richsalz**
Recreation of #2984; see some comments there.

### CODE REVIEW

File: test/bntest.c [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include <openssl/rand.h>
 #include "testutil.h"
 #include "test_main_custom.h"
+#include "test/testfile_helper.h"
```
Comment (levitte) [inline test/bntest.c]: Remove `test/`

Comment (richsalz) [inline test/bntest.c]: additional commit pushed.

**Comment by richsalz** [comment]
Is anything going to happen with this?  Or should I give up and just convert the existing bntest to use the framework?

**Comment by levitte** [comment]
Too many other things distracted me...  in the mean time, I've figured that there's one detail that I don't quite agree with...  it turns a read-one-stanza-at-the-time iterator into a eat-the-whole-file solution.  This might turn out to be a problem for anyone who wants to test this on a small memory device.

Otherwise, I'd welcome a general parser...  it will need some expansion if we want to have evp_test use it as well, but this does have value.

Sorry that it took so long to get back to this...

**Pull request closed without merging**